staging/src/k8s.io/apiserver/pkg/audit/request.go

```diff
@@ -204,3 +204,24 @@ func encodeObject(obj runtime.Object, gv schema.GroupVersion, serializer runtime
 	}
 	return nil, fmt.Errorf("no json encoder found")
 }
+
+// LogAnnotation fills in the Annotations according to the key value pair.
+func LogAnnotation(ae *auditinternal.Event, key, value string) {
+	if ae == nil || ae.Level.Less(auditinternal.LevelMetadata) {
+		return
+	}
+	if ae.Annotations == nil {
+		ae.Annotations = make(map[string]string)
+	}
+	ae.Annotations[key] = value
```

Comment:
@tallclair would expect an annotation to be able to be overwritten or not?

Reply:
Yeah, I think that's a good idea. I can see an argument for allowing annotations to be overwritten, e.g. it could simplify design a logging component to be able to overwrite the annotation (e.g. an earlier decision is overridden). But I think failing the overwrite is more robust, and could either indicate an error in the component code, or that the user is running 2 conflicting plugins.

Reply:
> I can see an argument for allowing annotations to be overwritten, e.g. it could simplify design a logging component to be able to overwrite the annotation (e.g. an earlier decision is overridden).

Aggre.
LogAnnotation is consistent with LogResponseObject, LogRequestPatch, I think.

Reply:
Done.